src/session.ts

```diff
@@ -282,54 +262,28 @@ export class SessionManager implements Middleware {
-
-            if (fixedPath !== powerShellExePath) {
-                const bitness = this.platformDetails.isOS64Bit ? 64 : 32;
-                // Show deprecation message with fix action.
-                // We don't need to wait on this to complete
-                // because we can finish gathering the configured
-                // PowerShell path without the fix
-                vscode
-                    .window
-                    .showWarningMessage(
-                        `The specified PowerShell path is incorrect for ${bitness}-bit VS Code, using '${fixedPath}' ` +
-                        "instead.",
-                        "Fix Setting Automatically")
-                    .then((choice) => {
-                        if (choice) {
-                            this.suppressRestartPrompt = true;
-                            Settings
-                                .change(
-                                    "powerShellExePath",
-                                    this.sessionSettings.developer.powerShellExePath,
-                                    true)
-                                .then(() => {
-                                    return Settings.change(
-                                        "developer.powerShellExePath",
-                                        undefined,
-                                        true);
-                                })
-                                .then(() => {
-                                    this.suppressRestartPrompt = false;
-                                });
-                        }
-                    });
-
-                powerShellExePath = fixedPath;
+        if (powerShellExePath) {
+            if (this.platformDetails.operatingSystem === OperatingSystem.Windows) {
+                // Check the path bitness
+                const fixedPath = this.powershellExeFinder.fixWindowsPowerShellPath(
+                    powerShellExePath);
+
+                if (fixedPath !== powerShellExePath) {
+                    // Show deprecation message with fix action.
+                    // We don't need to wait on this to complete
+                    // because we can finish gathering the configured
+                    // PowerShell path without the fix
+                    this.showBitnessPathFixWarning(fixedPath);
+                    powerShellExePath = fixedPath;
+                }
             }
+
+            return this.resolvePowerShellPath(powerShellExePath);
         }
 
-        return powerShellExePath.length > 0
-            ? this.resolvePowerShellPath(powerShellExePath)
-            : getDefaultPowerShellPath(this.platformDetails, this.sessionSettings.useX86Host);
+        // No need to resolve this path, since the finder guarantees its existence
+        const firstPowerShell = this.powershellExeFinder.getFirstAvailablePowerShellInstallation();
+        return firstPowerShell && firstPowerShell.exePath || null;
```

Comment:
The && and throwing me off here. Does that return `firstPowerShell` if `firstPowerShell` and `firstPowerShell.exePath` are truthy - else null?

Reply:
`&&` returns either the falsey argument, or if both are truthy, the rightmost.

So `"Hello" && "there"` evaluates to `"there'`.

So a common idiom I've seen is to use it like `obj?.property`

Reply:
oh I see. How about ternary instead? Maybe a little cleaner and more well known?
```suggestion
        return firstPowerShell ? firstPowerShell.exePath : null;
```

Reply:
Not as safe: if `exePath` is not defined, then the first will give a known default when the second does not

Reply:
Removed it with destructuring